server/src/main/java/org/candlepin/policy/SystemPurposeComplianceRules.java

```diff
@@ -140,19 +139,21 @@ public SystemPurposeComplianceStatus getStatus(Consumer consumer,
                 }
                 unsatisfiedAddons.removeAll(addonsFound);
 
-                if (product.hasAttribute(Product.Attributes.SUPPORT_LEVEL)) {
+                if (StringUtils.isNotEmpty(preferredSla) &&
```

Comment:
If there's something else to fix in this PR, we could also reorganize this code a bit to only do this check once. Both of the places it was added to could be nested within a single check, since both are contingent on it being non-empty.

Reply:
I'm not sure I understand.  One spot is `StringUtils.isNotEmpty(preferredSla)` and the other is `StringUtils.isNotEmpty(preferredUsage)`.

Reply:
Oof... disregard. This is what happens when I review code while dealing with migraine-induced double vision.